pkg/netpol/connlist/connlist.go

```diff
@@ -391,38 +416,47 @@ func (ca *ConnlistAnalyzer) getConnectionsList(pe *eval.PolicyEngine, ia *ingres
 	}
 	connsRes = append(connsRes, ingressAllowedConns...)
 
-	if len(peersAllowedConns) == 0 {
-		ca.logger.Warnf("connectivity analysis found no allowed connectivity between pairs from the configured workloads or external IP-blocks")
+	if ca.focusWorkload == "" && len(peersAllowedConns) == 0 {
+		ca.logger.Warnf("Connectivity analysis found no allowed connectivity between pairs from the configured workloads or external IP-blocks")
 	}
 
-	return connsRes, peersRes, nil
+	return connsRes, peers, nil
 }
 
-// getConnectionsBetweenPeers returns connections list from PolicyEngine object
-func (ca *ConnlistAnalyzer) getConnectionsBetweenPeers(pe *eval.PolicyEngine) ([]Peer2PeerConnection, []Peer, error) {
-	// get workload peers and ip blocks
-	peerList, err := pe.GetPeersList()
-	if err != nil {
-		ca.errors = append(ca.errors, newResourceEvaluationError(err))
-		return nil, nil, err
+// existsFocusWorkload checks if the provided focus workload is ingress-controller
+// or if it exists in the peers list from the parsed resources
+// if not returns a suitable warning message
+func (ca *ConnlistAnalyzer) existsFocusWorkload(peers []Peer, excludeIngressAnalysis bool) (existFocusWorkload bool, warning string) {
+	if ca.focusWorkload == ingressanalyzer.IngressPodName {
+		if excludeIngressAnalysis { // if the ingress-analyzer is empty,
+			// then no routes/k8s-ingress objects -> ingrss-controller pod will not be added
+			return false, "Pod ingress-controller was not added to the analysis, since Ingress/Route resources were not found."
+		}
```

Comment:
```suggestion
			return false, "The ingress-controller workload was not added to the analysis, since Ingress/Route resources were not found. Connectivity map report will be empty."
		}
```